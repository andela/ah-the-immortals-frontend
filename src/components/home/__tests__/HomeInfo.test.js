import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomePage from '../HomeInfo';

const props = {
  showModal: jest.fn(),
  show: false,
  closeModal: jest.fn(),
  handleChange: jest.fn(),
  handleSubmit: jest.fn(),
  handleSignInShow: jest.fn(),
  signInShow: false,
  handleSignInSubmit: jest.fn(),
  facebook: jest.fn(),
  google: jest.fn(),
  twitter: jest.fn(),
  handleSignInLink:jest.fn(),
  signupdata:{},
  errorShow:{},
  handleSignUpLink: jest.fn(),
  signindata: {},
  signInError: false
};

describe('Tests homepage component', () => {
  const wrapper = shallow(
    <HomePage {...props} />
  );
  it('tests the homepage component', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
