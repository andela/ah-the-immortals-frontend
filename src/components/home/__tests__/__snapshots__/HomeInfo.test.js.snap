// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests homepage component tests the homepage component 1`] = `
<div
  className="columns"
>
  <Bootstrap(Row)>
    <Col
      className="col"
    >
      <Container
        fluid={false}
      >
        <h1
          className="homepage"
        >
          Welcome to your articles Community
        </h1>
        <br />
        <Form
          inline={false}
        >
          <FormControl
            className="font-awesome"
            placeholder="ï€‚  Search for articles"
            type="text"
          />
        </Form>
      </Container>
      <Container
        className="home"
        fluid={false}
      >
        <p>
          <b>
            Who is Authors Haven for?
          </b>
        </p>
        <p>
          Anyone looking to inspire and be inspired by ideas
        </p>
        <p>
          Join fellow authors and enthusiasts in Authors Haven
        </p>
        <Button
          active={false}
          disabled={false}
          onClick={[MockFunction]}
          type="button"
          variant="primary"
        >
          Get Started
        </Button>
        <SignupModal
          closeModal={[MockFunction]}
          errorShow={Object {}}
          facebook={[MockFunction]}
          google={[MockFunction]}
          handleChange={[MockFunction]}
          handleSignInLink={[MockFunction]}
          handleSubmit={[MockFunction]}
          show={false}
          signupdata={Object {}}
          twitter={[MockFunction]}
        />
        <LogInModal
          closeModal={[MockFunction]}
          facebook={[MockFunction]}
          google={[MockFunction]}
          handleChange={[MockFunction]}
          handleSignInSubmit={[MockFunction]}
          signInShow={false}
          twitter={[MockFunction]}
        />
      </Container>
    </Col>
    <Col
      className="homepageimg"
    >
      <img
        alt=""
        height="400"
        src="test-file-stub"
        width="450"
      />
    </Col>
  </Bootstrap(Row)>
</div>
`;
